packages/block-library/src/query/edit/query-inspector-controls.js

```diff
@@ -143,8 +150,24 @@ export default function QueryInspectorControls( {
 		onChangeDebounced();
 		return onChangeDebounced.cancel;
 	}, [ querySearch, onChangeDebounced ] );
+
+	const CreateNewPostLink = ( { type } ) => {
+		const newPostUrl = addQueryArgs( 'post-new.php', {
+			post_type: type,
+		} );
+		return (
+			<div className="wp-block-query__create-new-link">
+				{ createInterpolateElement(
+					__( '<a>Create a new post</a> for this feed.' ),
+					// eslint-disable-next-line jsx-a11y/anchor-has-content
+					{ a: <a href={ newPostUrl } /> }
+				) }
+			</div>
+		);
+	};
 	return (
 		<InspectorControls>
+			<CreateNewPostLink type={ postType } />
```

Comment:
I would either:

* inline the whole element tree, i.e. `<InspectorControls><div className="...create-new-link">...`
* or move the definition of `CreateNewPostLink` one level up, to the module

Reply:
I moved it one level up.